### PR TITLE
[CosmosSDK] Bump to v0.50.7

### PR DESCRIPTION
**Pull request by okdas**
## Summary

Bumps cosmos-sdk to the latest available.

## Issue

- #591 

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

**Documentation changes** (only if making doc changes)
- [ ] `make docusaurus_start`; only needed if you make doc changes

**Local Testing** (only if making code changes)
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- See [quickstart guide](https://dev.poktroll.com/developer_guide/quickstart) for instructions

**PR Testing** (only if making code changes)
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.
    - **THIS IS VERY EXPENSIVE**, so only do it after all the reviews are complete.
    - Optionally run `make trigger_ci` if you want to re-trigger tests without any code changes
    - If tests fail, try re-running failed tests only using the GitHub UI as shown [here](https://github.com/pokt-network/poktroll/assets/1892194/607984e9-0615-4569-9452-4c730190c1d2)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [ ] I have commented my code
- [ ] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

## Summary by CodeRabbit

- **New Features**
  - Updated module dependencies to the latest versions for improved performance and security.

- **Configuration Changes**
  - Mempool configuration now allows an unlimited number of transactions.
  - Updated version in `config.toml` to "0.38.7".

### CODE REVIEW

**Comment by okdas** [comment]
Alright, everything seems to work locally, including e2e and stress tests. I had to bump my local `buf` binary to the latest. With the version changed in `go.mod`, it might not be necessary, but this is a note for the reviewer to try to switch to that branch and run a LocalNet. The new developer setup should work; it's the already existing setup I'm worried about.

**Comment by okdas** [comment]
Screenshot 2024-07-10 at 6 46 05 PM

@bryanchriswhite do you have any cycles to look into that? jk, the PR is not that large! Most of the stuff is code generated.

File: localnet/poktrolld/config/app.toml [hunk 1]
```diff
@@ -32,7 +32,7 @@ query-gas-limit = "0"
   enable = true
 
 [mempool]
-  max-txs = 5000
+  max-txs = -1
```
Comment (bryanchriswhite) [inline localnet/poktrolld/config/app.toml:35]: Is this intentional?

According to [the docs](https://docs.cosmos.network/main/learn/advanced/config), this disables the mempool but I don't see why we would want to do that.

**Review by bryanchriswhite — COMMENTED** [review]
:rocket:

**Review by bryanchriswhite — APPROVED** [review]
Thanks @okdas! :raised_hands:  

Preemptively approving. I just wanted to confirm the mempool config change.

File: localnet/poktrolld/config/app.toml (hunk 1 above)
Comment (okdas) [reply localnet/poktrolld/config/app.toml:35]: @bryanchriswhite this is the value cosmos-sdk generates by default now. I, personally, like to keep the default values. And we do adjust this config on the live network.  

Related PRs explaining the change:

https://github.com/cosmos/cosmos-sdk/pull/20008 (original: https://github.com/cosmos/cosmos-sdk/pull/19970)
https://github.com/cosmos/cosmos-sdk/pull/20012

Comment (okdas) [reply localnet/poktrolld/config/app.toml:35]: @bryanchriswhite  To be clear, I do not oppose using the `5000` value here! :) Reading these two PRs still leaves the question of what the expectation for this value is on the full-blown network. I guess, just like with the existing mainnet, we will learn the behavior of the network and adjust.

**Pull request merged**